src/main/controllers/CourtDetailsController.ts

```diff
@@ -0,0 +1,59 @@
+import { FactRequest } from '../interfaces/FactRequest';
+import { Response } from 'express';
+import { FactApi } from '../utils/FactApi';
+import { isEmpty } from '../utils/validation';
+import autobind from 'autobind-decorator';
+import { Enquiries } from '../interfaces/Enquiries';
+import { CourtDetailsData } from '../interfaces/CourtDetailsData';
+import config from 'config';
+
+@autobind
+export class CourtDetailsController {
+
+  //TODO this comes into place when the user comes from a no name court search
+  private regionalCentre = false;
+
+  constructor(
+      private readonly api: FactApi
+  ) { }
+
+  public async get(req: FactRequest, res: Response) {
+    const slug: string = req.params.slug as string;
+    const data: CourtDetailsData = {
+      ...req.i18n.getDataByLanguage(req.lng)['court-details'],
+      path: '/individual-location-pages/courts',
+      results: [],
+      errors: false,
+    };
+
+    if (!isEmpty(slug)) {
+      const courts: any = await this.api.court(slug);
```

Comment:
If you properly typed this you wouldn't have so much noise on the lines below

Reply:
can types handle funcitons?

Reply:
it can be you don't need to, you are specifying the type of `const courts` which has the return type of the function after you've awaited the promise.